tests/ext/dogstatsd/metrics_host_config.phpt

```diff
@@ -0,0 +1,34 @@
+--TEST--
+Test DogStatsD configuration with DD_DOGSTATSD_HOST and DD_DOGSTATSD_PORT
+--SKIPIF--
+<?php if (!extension_loaded('sockets')) die('skip: the sockets extension is required for this test'); ?>
+--ENV--
+DD_DOGSTATSD_HOST=192.168.1.1
+DD_DOGSTATSD_PORT=9876
+--FILE--
+<?php
+class MockUDPSocket {
+    public function stream_open($path, $mode, $options, &$opened_path) {
+        $parts = parse_url($path);
+        if (isset($parts['host']) && isset($parts['port'])) {
+            echo "DogStatsD client attempted to connect to: " . $parts['host'] . ":" . $parts['port'] . "\n";
+        }
+        return true;
+    }
+
+    public function stream_write($data) {
+        return strlen($data);
+    }
+
+    public function stream_close() {
+        return true;
+    }
+}
+
+// Register our mock socket wrapper for udp connections
+stream_wrapper_register('udp', 'MockUDPSocket');
```

Comment:
This does not work for connections initiated from the C code. You cannot mock this; you'll need an actual socket.